firmware-management/swupdate-handlers/arm-handler-common.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include <mntent.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include "util.h"
+#include <arm-handler-common.h>
+
+char *str_new(const size_t size)
+{
+    char *dst;
+    if (!(dst = malloc(sizeof dst * size)))
+    {
+        ERROR("%s", "Failed to allocate string");
+        return NULL;
+    }
+    return dst;
+}
+
+void str_delete(char *str)
+{
+    free(str);
+}
+
+char *str_copy(const char *src)
```

Comment:
This function is quite different to `strcpy`, so I'd probably call it something like `str_copy_to_new` to make it clear that it allocates.

Reply:
Could make `src` const.
```suggestion
char *str_copy_to_new(const char * const src)
```

Reply:
3b4aa252c3ff84deb0cd6de42c6f83654601b2b6